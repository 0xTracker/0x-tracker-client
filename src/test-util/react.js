import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import React from 'react';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

export { renderWithRouter };
