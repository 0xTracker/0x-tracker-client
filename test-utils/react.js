import { createMemoryHistory } from 'history';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from 'react-testing-library';
import { Router } from 'react-router-dom';
import React from 'react';

import reducer from '../src/redux/reducer';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

const renderWithAppContext = (
  component,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState,
    store = createStore(reducer, initialState),
  } = {},
) => ({
  ...render(
    <Provider store={store}>
      <Router history={history}>{component}</Router>
    </Provider>,
  ),
  history,
});

export { renderWithAppContext, renderWithRouter };
