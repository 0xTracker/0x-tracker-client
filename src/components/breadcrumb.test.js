import React from 'react';

import { renderWithRouter } from '../test-util/react';
import Breadcrumb from './breadcrumb';

it('should render with items', () => {
  const { container } = renderWithRouter(
    <Breadcrumb
      items={[
        { title: 'Tokens', url: '/tokens' },
        { title: 'Ether', url: '/tokens/eth' },
      ]}
    />,
  );

  expect(container.firstChild).toMatchSnapshot();
});
