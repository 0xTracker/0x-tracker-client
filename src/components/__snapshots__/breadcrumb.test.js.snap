// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with items 1`] = `
<nav
  aria-label="breadcrumb"
  class="breadcrumb__StyledBreadcrumb-yqtjg0-0 eCKXTH"
>
  <ol
    class="breadcrumb"
  >
    <li
      class="breadcrumb-item"
    >
      <a
        class="link__StyledInternalLink-sc-1m1n17q-1 jZaAPW"
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="breadcrumb-item"
    >
      <a
        class="link__StyledInternalLink-sc-1m1n17q-1 jZaAPW"
        href="/tokens"
      >
        Tokens
      </a>
    </li>
    <li
      aria-current="page"
      class="active breadcrumb-item"
    >
      Ether
    </li>
  </ol>
</nav>
`;
